crates/starknet-types-core/src/felt/mod.rs

```diff
@@ -50,20 +52,41 @@ pub struct Felt(pub(crate) FieldElement<Stark252PrimeField>);
 #[derive(Debug, Clone, Copy)]
 pub struct NonZeroFelt(FieldElement<Stark252PrimeField>);
 
+impl NonZeroFelt {
+    /// Create a [NonZeroFelt] as a constant.
+    /// # Safety
+    /// If the value is zero will panic.
+    pub const fn from_raw(value: [u64; 4]) -> Self {
+        let mut i = 0;
+        let mut zeros_nb = 0;
+        while i < value.len() {
+            if value[i] == 0 {
+                zeros_nb += 1;
+            }
+            i += 1;
+        }
+        assert!(zeros_nb < value.len(), "Felt is zero");
```

Comment:
are we sure about that? 
I don't know much about Montgomery, can you explain ?

Reply:
if everything is 0 then the value is 0